x-pack/plugins/reporting/export_types/csv_from_savedobject/server/lib/generate_csv_search.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { buildEsQuery } from '@kbn/es-query';
+import { Request } from 'hapi';
+import moment from 'moment';
+import { KbnServer, Logger } from '../../../../types';
+// @ts-ignore
+import { fieldFormatMapFactory } from '../../../csv/server/lib/field_format_map';
+// @ts-ignore
+import { createGenerateCsv } from '../../../csv/server/lib/generate_csv';
+import {
+  IndexPatternSavedObject,
+  SavedSearchObjectAttributes,
+  SearchPanel,
+  SearchRequest,
+  SearchSource,
+  TimeRangeParams,
+} from '../../types';
+import { getDataSource } from './get_data_source';
+
+interface SavedSearchGeneratorResult {
+  content: string;
+  maxSizeReached: boolean;
+  size: number;
+}
+
+interface CsvResultFromSearch {
+  type: string;
+  result: SavedSearchGeneratorResult | null;
+}
+
+type EndpointCaller = (method: string, params: any) => Promise<any>;
+type FormatsMap = Map<
+  string,
+  {
+    id: string;
+    params: {
+      pattern: string;
+    };
+  }
+>;
+
+interface GenerateCsvParams {
+  searchRequest: SearchRequest;
+  callEndpoint: EndpointCaller;
+  fields: string[];
+  formatsMap: FormatsMap;
+  metaFields: string[]; // FIXME not sure what this is for
+  conflictedTypesFields: string[]; // FIXME not sure what this is for
+  cancellationToken: any; // FIXME not sure how to do anything with this
```

Comment:
If the search is taking longer than the reporting timeout config, it should be automatically cancelled. But this might be more of a TODO for a later phase. Thoughts?